StepperMotor/StepperMotor.c

```diff
@@ -161,15 +164,18 @@ void TIMER1_IRQHandler(void)
 	// case None defined within the StepperMotor_Step* functions when adminstration is done
 	switch(Control_GlobalStatus)
 	{
-		case Basal_Status: 
+		case Basal_Status:
+      BasalDose_TimingDisable();
 			StepperMotor_CurrentBasalDose++; // Keep track of current dosing
 			StepperMotor_StepForward();
 			break;
 		case Bolus_Status:
+      BasalDose_TimingDisable();
 			StepperMotor_CurrentBolusDose++; 
 			StepperMotor_StepForward();
 			break;
 		case Backward_Status:
+      BasalDose_TimingDisable();
```

Comment:
In the StepperMotor_SpinEnable, the first line that function is BasaDose_TimingDisable. So, it is  already disabling Timer0.  So it shouldn't be triggering multiple interrupts. But I moved the disable into the main loop in the in the switch statments for empty and full